src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -28,9 +28,14 @@ def _resolve_strict_dependencies(target):
 
 def _resolve_exports(target):
   for export in getattr(target, 'exports', []):
-    for exp in _resolve_exports(export):
-      yield exp
-    yield export
+    if type(export) in (AliasTarget, Target):
+      # If exported target is an alias, expand its dependencies.
+      for dep in _resolve_strict_dependencies(export):
```

Comment:
Could you add a test case for this bit of mutual recursion? It does make sense that it would work this way, but I'd like to see some examples.